pkg/utils/arrays.go

```diff
@@ -1,7 +1,22 @@
 package utils
 
+import (
+	"fmt"
+)
+
 func Reverse[T any](a []T) {
 	for i, j := 0, len(a)-1; i < j; i, j = i+1, j-1 {
 		a[i], a[j] = a[j], a[i]
 	}
 }
+
+func Pop[T any](a *[]T) (T, error) {
+	if len(*a) == 0 {
+		var zeroValue T
```

Comment:
@MaksymMalicki I had it like 
![image](https://github.com/NethermindEth/cairo-vm-go/assets/11419017/f51e8241-9463-43ad-8d14-394ffbad970f)

But it is giving me a warning
If you want I can just ignore it
![image](https://github.com/NethermindEth/cairo-vm-go/assets/11419017/82e434cb-bb79-4367-b613-d5e1483f1d65)

I was also considering make the return type a pointer `*T` but not sure if that's the best approach

What do you think?

Reply:
No no, its okey, I was just wondering haha 😛. Returning pointer is not a good idea, as it will cause heap allocations and performance issues. The current solution looks great

Reply:
@MaksymMalicki ok cool
Thanks for the explanation